src/device_constructors/renewablegeneration_constructor.jl

```diff
@@ -0,0 +1,95 @@
+function construct_device!(ps_m::CanonicalModel,
```

Comment:
Should this function have a parameter for the type of sys.generators.renewable? The logic is completely different for PSY.RenewableFix vs !PSY.RenewableFix.